components/renku_data_services/storage_adapters/core.py

```diff
@@ -46,7 +44,11 @@ async def get_storage(
                 stmt = stmt.where(schemas.CloudStorageORM.name == name)
             res = await session.execute(stmt)
             orms = res.scalars().all()
-            return [orm.dump() for orm in orms if orm.project_id == user.project_id]
+            accessible_projects = await user.filter_projects_by_access_level(
+                [p.project_id for p in orms], base_models.GitlabAccessLevel.MEMBER
+            )
+
+            return [p.dump() for p in orms if p.project_id in accessible_projects]
```

Comment:
Also it would be nice if you did not have to pull ALL user projects from gitlab when a specific project ID is passed in. In this case you just want to see if the access level for that project ID is high enough and that is it. Which is probably much faster than pulling in all projects for a user.

Reply:
what do you mean? it doesn't pull all a users projects from gitlab, only the ones passed by id.
And it does so with the GraphQL endpoint that can do it in one go (up to pagination limit)

Reply:
And we can't get postgres to filter because we need to know the relevant project ids first, which we get from postgres.
Unless you're suggesting to do two roundtrips to postgres?